### PR TITLE
mptcp: initial disconnect coverage

### PR DESCRIPTION
**Pull request by pabeni**
Add basic test for connect(AF_UNSPEC) for a newly accepted
socket.

Signed-off-by: Paolo Abeni <pabeni@redhat.com>

### CODE REVIEW

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
++0.0  socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0.0  setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
++0.0  fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0  fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0    bind(3, ..., ...) = 0
++0    listen(3, 1) = 0
+
++0.0    <  addr[caddr0] > addr[saddr0]  S   0:0(0)         win 65535  <mss 1460, sackOK, TS val 4074410674 ecr 0,          nop, wscale 8, mpcapable v1 flags[flag_h] nokey>
```
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt]: Small detail: do we need to specify the addresses? `addr[caddr0] > addr[saddr0]`

**Review by matttbe — COMMENTED** [review]
Thank you for sharing this!

Just one small comment, the rest looks good to me!
I can do the modif if you prefer.

**Review by dcaratti — APPROVED** [review]
LGTM!

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
++0.0  setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
++0.0  fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0  fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0    bind(3, ..., ...) = 0
++0    listen(3, 1) = 0
+
++0       <  S   0:0(0)         win 32792  <mss 1000, sackOK, nop, nop, nop, wscale 7, mpcapable v1 flags[flag_h] nokey>
++0       >  S.  0:0(0)  ack 1             <mss 1460, nop, nop, sackOK, nop, wscale 8, mpcapable v1 flags[flag_h] key[skey]>
++0.01    <   .  1:1(0)  ack 1  win 257                                               <mpcapable v0 flags[flag_h] key[ckey=2, skey]>
```
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt]: (as discussed on IRC: you switched to v0 here)

File: gtests/net/mptcp/syscalls/disconnect_after_accept.pkt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
++0.0  socket(..., SOCK_STREAM, IPPROTO_MPTCP) = 3
++0.0  setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
++0.0  fcntl(3, F_GETFL) = 0x2 (flags O_RDWR)
++0.0  fcntl(3, F_SETFL, O_RDWR|O_NONBLOCK) = 0
+
++0    bind(3, ..., ...) = 0
++0    listen(3, 1) = 0
+
++0       <  S   0:0(0)         win 32792  <mss 1000, sackOK, nop, nop, nop, wscale 7, mpcapable v1 flags[flag_h] nokey>
```
Comment (matttbe) [inline gtests/net/mptcp/syscalls/disconnect_after_accept.pkt:13]: (not sure we need mss 1000 but it doesn't hurt)

**Review by matttbe — CHANGES_REQUESTED** [review]

**Review by matttbe — APPROVED** [review]

**Comment by matttbe** [comment]
We can merge this once Paolo's kernel patch is merged in our tree.

I tested the new .pkt on top of the export branch, with and without Paolo's fix. I can reproduce the issue without the fix.

**Pull request merged**